integration-test/background/storage.js

```diff
@@ -19,73 +17,111 @@ describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, (
             },
             { polling: 100, timeout: 10000 }
         )
+        return { browser, page }
+    }
+
+    describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, () => {
+        let cookies = []
 
-        let iframeFullyLoaded = false
-        try {
-            page.on('requestfinished', (req) => {
-                // once we see this url, we can consider the test completed
-                if (req.url().startsWith(`https://${thirdPartyTracker}/set-cookie`)) {
-                    iframeFullyLoaded = true
+        beforeAll(async () => {
+            let iframeFullyLoaded = false
+            const { browser, page } = await setup()
+            try {
+                page.on('requestfinished', (req) => {
+                    // once we see this url, we can consider the test completed
+                    if (req.url().startsWith(`https://${thirdPartyTracker}/set-cookie`)) {
+                        iframeFullyLoaded = true
+                    }
+                })
+                // Load the test pages home first to give some time for the extension background to start
+                // and register the content-script-message handler
+                await page.goto(`https://${testPageDomain}/`, { waitUntil: 'networkidle0' })
+                await page.bringToFront()
+                await page.goto(`https://${testPageDomain}/privacy-protections/storage-blocking/?store`, { waitUntil: 'networkidle0' })
+                await page.bringToFront()
+                // eslint-disable-next-line no-unmodified-loop-condition
+                while (!iframeFullyLoaded) {
```

Comment:
`page.waitForRequest` or `page.waitForResponse` will likely be much less flakey and make the code simpler.

Reply:
It seems to now work using only `waitUntil: 'networkidle2'`.